modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/product_stack.py

```diff
@@ -210,6 +223,23 @@ def __init__(
                     self,
                     "Policy",
                     statements=[
+                        iam.PolicyStatement(
+                            actions=[
+                                "codebuild:ImportSourceCredentials",
+                                "codebuild:DeleteSourceCredentials",
+                                "codebuild:ListSourceCredentials",
+                            ],
+                            resources=["*"],
+                        ),
+                        iam.PolicyStatement(
+                            actions=[
+                                "codeconnections:UseConnection",
+                                "codeconnections:PassConnection",
+                                "codeconnections:GetConnection",
+                                "codeconnections:GetConnectionToken",
+                            ],
+                            resources=[f"arn:aws:codeconnections:*:{Aws.ACCOUNT_ID}:connection/*"],
```

Comment:
Given we have full code connection arn, can we scope the resource to the specific code connection resource?

Reply:
Actually, connection arn contains last part as identifier with random string hence I considered adding star. But I agree with you to restrict permission specific to connection that is required. I check on this and update. Thanks for pointing this out @kukushking